projects/fal/src/fal/api.py

```diff
@@ -791,11 +794,26 @@ class BaseServable:
     def collect_routes(self) -> dict[RouteSignature, Callable[..., Any]]:
         raise NotImplementedError
 
-    def _add_extra_middlewares(self, app: FastAPI):
+    def _add_extra_middlewares(self, app: FastAPI) -> None:
         """
         For subclasses to add extra middlewares to the app.
         """
-        pass
+
+        # Common middlewares
+        @app.middleware("http")
+        async def fal_logger_middleware(request, call_next):
+            try:
+                fal_headers = {
+                    header: request.headers[header]
+                    for header in request.headers
+                    if header.startswith("X-Fal-Log-")
+                }
+                # TODO: somehow add the header to the logger context of the served function or app
```

Comment:
how would this work with multiplexed? I think we can use something like a contextvar to have a RequestState https://docs.python.org/3/library/contextvars.html and store stuff like `request_user`, `request_user_upload_preference`, `request_id`, etc.